File: src/types/task.ts
export type TaskStatus = 'backlog' | 'todo' | 'in_progress' | 'cooking' | 'paused' | 'review' | 'done' | 'failed'
export type TaskPriority = 'low' | 'medium' | 'high'
export type TaskType = 'generated' | 'custom'

export interface Task {
  id: string
  deckId: string
  type?: TaskType
  taskType?: TaskType
  name: string
  description?: string
  status: TaskStatus
  priority: TaskPriority
  planId?: string
  phaseId?: string
  worktreeId?: string
  worktreeName?: string
  cookSessionId?: string
  createdAt?: string
  updatedAt?: string
}

export interface TaskColumn {
  id: TaskStatus
  title: string
  tasks: Task[]
}

File: src-tauri/src/models/task.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum TaskStatus {
    Backlog,
    Todo,
    Cooking,
    Paused,
    Review,
    Done,
    Failed,
}

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum TaskPriority {
    Low,
    Medium,
    High,
}

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum TaskType {
    Generated,
    Custom,
}

#[derive(Serialize, Deserialize, Clone, Debug)]
#[serde(rename_all = "camelCase")]
pub struct Task {
    pub id: String,
    pub deck_id: String,
    pub task_type: TaskType,
    pub name: String,
    pub description: Option<String>,
    pub status: TaskStatus,
    pub priority: TaskPriority,
    pub plan_id: Option<String>,
    pub phase_id: Option<String>,
    pub worktree_id: Option<String>,
    pub cook_session_id: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}
